commander/new_block.go

```diff
@@ -153,6 +153,11 @@ func (c *Commander) syncForward(latestBlockNumber uint64) (*uint64, error) {
 		return nil, err
 	}
 
+	err = c.withdrawRemainingStakes(startBlock, endBlock)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
This needs to happen **after** we are synced to the latest mined block. Before that we cannot tell whether we have already withdrawn or not. We could be sending redundant withdrawal txs here and loose Ether on fees.